### PR TITLE
Propagate inverse beta variable to pat muons (muon POG)

### PR DESCRIPTION
**Pull request by cericeci**
#### PR description:

The PR propagates inverse beta timing information to be available for pat::Muon.

We would like to backport it later to 10_6_X (adapted to run only in 2016 UL conditions). 

@peruzzim @Fedespring 

#### PR validation:
No compilation errors or of battery of tests from
runTheMatrix.py -l limited -i all

Unit testing with [pat::Muon only producer](https://github.com/cms-sw/cmssw/blob/master/PhysicsTools/PatAlgos/test/patTuple_onlyMuons_cfg.py) reproduces values stored at RECO level

### CODE REVIEW

**Comment by slava77** [comment]
@cericeci 
without code-checks passing this PR can not be tested.
please address https://github.com/cms-sw/cmssw/pull/28212#issuecomment-543819137
Thank you.

**Comment by perrotta** [comment]
please test

File: DataFormats/PatCandidates/interface/Muon.h [hunk 1]
```diff
@@ -32,400 +33,413 @@
...
+    reco::TrackRef tpfmsMuon() const { return tpfmsTrack(); }  // JMTBAD gcc deprecated attribute?
+    /// embed reference to the above picky Track
+    void embedPickyMuon();
+    /// embed reference to the above tpfms Track
+    void embedTpfmsMuon();
+    /// embed reference to the above dyt Track
+    void embedDytMuon();
+
+    // add extra timing information
+    void readExtraTimerInfo(const reco::MuonTimeExtra& t);
```
Comment (slava77) [inline DataFormats/PatCandidates/interface/Muon.h]: ```readTimeExtra``` may be more clear; "timer" is confusing with this word usually used for execution or wall clock time counting.

this method is redundant with setInverseBeta and setInverseBetaErr.
perhaps it's more appropriate to remove setInverseBeta and setInverseBetaErr, in a case there is some need to expand beyond just beta and betaErr.

File: DataFormats/PatCandidates/interface/Muon.h [hunk 2]
```diff
@@ -32,400 +33,413 @@
...
+    // Low pt Muon MVA
+    float lowptMvaValue() const { return lowptMvaValue_; }
+    void setLowPtMvaValue(float lowptmva) { lowptMvaValue_ = lowptmva; }
+
+    /// Soft Muon MVA
+    float softMvaValue() const { return softMvaValue_; }
+    void setSoftMvaValue(float softmva) { softMvaValue_ = softmva; }
+
+    /// Inverse beta
+    void setInverseBeta(const float iBeta) { inverseBeta_ = iBeta; };
```
Comment (slava77) [inline DataFormats/PatCandidates/interface/Muon.h]: perhaps copy the rest of the comment in the MuonTimeExtra
``` for prompt particle hypothesis (time is constrained to the bunch crossing time)```

File: DataFormats/PatCandidates/interface/Muon.h [hunk 3]
```diff
@@ -32,400 +33,413 @@
...
+    void setInverseBetaErr(const float iBetaErr) { inverseBetaErr_ = iBetaErr; };
+    float inverseBeta() const { return inverseBeta_; };
+    float inverseBetaErr() const { return inverseBetaErr_; };
```
[the remark this answers is not in the document]
Comment (cericeci) [reply DataFormats/PatCandidates/interface/Muon.h]: Opted to remove the unused "setters" as per your suggestion below

**Comment by cericeci** [comment]
Updated following @slava77 suggestions

**Comment by slava77** [comment]
@cmsbuild please test

File: DataFormats/PatCandidates/interface/Muon.h [hunk 4]
```diff
@@ -129,7 +129,9 @@ namespace pat {
     void embedDytMuon();
 
     // add extra timing information
-    void readExtraTimerInfo(const reco::MuonTimeExtra& t);
+    /// 1/beta for prompt particle hypothesis
+    /// (time is constraint to the bunch crossing time)
```
Comment (slava77) [inline DataFormats/PatCandidates/interface/Muon.h]: please move this comment to the accessor (around L298).
The readTime\* method is here to encapsulate whatever else can be read and the details of 1/beta here are not the most relevant.

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/28212  010a58f 
- code changes are in line with the PR description and the follow up review
- jenkins tests pass and comparisons with the baseline show no differences (the new variables were not yet added to the monitoring setup).

**Comment by santocch** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Comment by fabiocos** [comment]
@peruzzim this is an example of development PR where a data format changes in between two development pre-releases...

**Pull request merged**